src/tools/std20/views/test/iota.test.cpp

```diff
@@ -0,0 +1,145 @@
+// include Catch2
+#include <catch2/catch_test_macros.hpp>
+
+// what we are testing
+#include "tools/std20/iterator.hpp"
+#include "tools/std20/views.hpp"
+
+// other includes
+#include "tools/std20/algorithm.hpp"
+
+// convenience typedefs
+//using namespace njoy::tools;
+namespace std20 = nano;
+
+SCENARIO( "iota_view" ) {
+
+  const std::vector< int > equal = { 1, 2, 3, 4, 5 };
+
+  GIVEN( "a value" ) {
+
+    WHEN( "creating an iota view" ) {
+
+      auto chunk = std20::ranges::iota_view< int, int >( 1 );
+      using Range = decltype( chunk );
+      using Iterator = nano::iterator_t< Range >;
+
+      THEN( "the transform_view satisfies the required concepts" ) {
+
+		    CHECK( std20::ranges::view< Range > );
+		    CHECK( std20::ranges::sized_range < Range > );
+        CHECK( std20::ranges::forward_range< Range > );
+        CHECK( std20::ranges::bidirectional_range< Range > );
+		    CHECK( std20::ranges::random_access_range < Range > );
+		    CHECK( ! std20::ranges::contiguous_range < Range > );
+		    CHECK( std20::ranges::common_range < Range > );
+      }
+
+      THEN( "the transform_view range and iterator associated types are correct" ) {
+
+        CHECK( std20::same_as< nano::ranges::range_value_t< Range >, int > );
+        CHECK( std20::same_as< nano::ranges::range_reference_t< Range >, int > );
+        CHECK( std20::same_as< nano::ranges::range_difference_t< Range >, int > );
+
+        CHECK( std20::same_as< nano::ranges::iter_value_t< Iterator >, int > );
+        CHECK( std20::same_as< nano::ranges::iter_reference_t< Iterator >, int > );
+        CHECK( std20::same_as< nano::ranges::iter_difference_t< Iterator >, int > );
```

Comment:
should the `_reference_t<>` and `_difference_t<>` be compared with `int&` and `std::ptrdiff_t`, respectively?

Likewise for other THENs in scenario?

Reply:
These are particular cases: sometimes you cannot change the value in a view, as is the case here for iota. If you test reference_t<> against int&, the test would fail.